File: frameNet/src/main/java/org/sqlunet/framenet/browser/FnSentenceFragment.java
/*
 * Copyright (c) 2019. Bernard Bou <1313ou@gmail.com>.
 */

package org.sqlunet.framenet.browser;

import android.os.Bundle;
import android.os.Parcelable;

import org.sqlunet.browser.Module;
import org.sqlunet.browser.TreeFragment;
import org.sqlunet.framenet.R;
import org.sqlunet.framenet.loaders.SentenceModule;
import org.sqlunet.provider.ProviderArgs;
import org.sqlunet.treeview.model.TreeNode;

import androidx.annotation.Nullable;

/**
 * A fragment representing a sentence
 *
 * @author <a href="mailto:1313ou@gmail.com">Bernard Bou</a>
 */
public class FnSentenceFragment extends TreeFragment
{
	// static private final String TAG = "FnSentenceF";

	/**
	 * Constructor
	 */
	public FnSentenceFragment()
	{
		this.layoutId = R.layout.fragment_fnsentence;
		this.treeContainerId = R.id.data_contents;
		this.headerId = R.string.framenet_sentences;
		this.iconId = R.drawable.sentence;
	}

	@Override
	public void onCreate(@Nullable final Bundle savedInstanceState)
	{
		super.onCreate(savedInstanceState);

		// query
		final Bundle args = getArguments();
		assert args != null;
		final int type = args.getInt(ProviderArgs.ARG_QUERYTYPE);
		if (args.containsKey(ProviderArgs.ARG_QUERYPOINTER))
		{
			// pointer
			final Parcelable pointer = args.getParcelable(ProviderArgs.ARG_QUERYPOINTER);

			// root node
			final TreeNode queryNode = this.treeRoot.getChildren().iterator().next();

			// module
			final Module module = new SentenceModule(this);
			module.init(type, pointer);
			module.process(queryNode);
		}
	}
}

File: frameNet/src/main/java/org/sqlunet/framenet/browser/FnSentenceFragment.kt
/*
 * Copyright (c) 2023. Bernard Bou
 */
package org.sqlunet.framenet.browser

import android.os.Bundle
import org.sqlunet.browser.Module
import org.sqlunet.browser.TreeFragment
import org.sqlunet.framenet.R
import org.sqlunet.framenet.loaders.SentenceModule
import org.sqlunet.provider.ProviderArgs

/**
 * A fragment representing a sentence
 *
 * @author [Bernard Bou](mailto:1313ou@gmail.com)
 */
class FnSentenceFragment : TreeFragment() {

    init {
        layoutId = R.layout.fragment_fnsentence
        treeContainerId = R.id.data_contents
        headerId = R.string.framenet_sentences
        iconId = R.drawable.sentence
    }

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)

        // query
        val args = requireArguments()
        val type = args.getInt(ProviderArgs.ARG_QUERYTYPE)
        val standAlone = args.getBoolean(ProviderArgs.ARG_STANDALONE)
        if (args.containsKey(ProviderArgs.ARG_QUERYPOINTER)) {
            // pointer
            val pointer = getPointer(args)

            // root node
            val queryNode = treeRoot.children.iterator().next()

            // module
            val module: Module = SentenceModule(this, standAlone)
            module.init(type, pointer)
            module.process(queryNode)
        }
    }
}
